src/mds/MDLog.cc

```diff
@@ -168,6 +168,22 @@ uint64_t MDLog::get_safe_pos() const
   return journaler->get_write_safe_pos(); 
 }
 
+estimated_reply_time_t MDLog::get_replay_estimated_complete_time(){
+  auto current_pos = journaler->get_read_pos();
+  auto total_bytes = end_pos - start_pos;
+  estimated_reply_time_t estimated_time;
+  double percent_complete = (double)  (current_pos - start_pos) / total_bytes;
+  auto time = percent_complete * \
+    std::chrono::duration_cast<std::chrono::seconds>(ceph::coarse_mono_clock::now() - start_time);
```

Comment:
Test for the correctness of this estimate. You'll see this calculation is not correct.